From a408321154d45342a42f4b54e44af7676b4c4933 Mon Sep 17 00:00:00 2001
From: Nicolas Adenis-Lamarre <nicolas.adenis.lamarre@gmail.com>
Date: Wed, 28 Jun 2023 05:23:16 +0000
Subject: [PATCH] close correctly the uninitialized joysticks

without that closing, the number of open files is growing (can be observed via lsof | grep kodi | grep /dev)
and ends to crash kodi after a long time

Signed-off-by: Nicolas Adenis-Lamarre <nicolas.adenis.lamarre@gmail.com>
---
 src/api/udev/JoystickInterfaceUdev.cpp | 9 +++++++--
 src/api/udev/JoystickUdev.h            | 1 +
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/api/udev/JoystickInterfaceUdev.cpp b/src/api/udev/JoystickInterfaceUdev.cpp
index 8ee6dfa3..0459927c 100644
--- a/src/api/udev/JoystickInterfaceUdev.cpp
+++ b/src/api/udev/JoystickInterfaceUdev.cpp
@@ -92,8 +92,13 @@ bool CJoystickInterfaceUdev::ScanForJoysticks(JoystickVector& joysticks)
 
      if (devnode != nullptr)
      {
-       JoystickPtr joystick = JoystickPtr(new CJoystickUdev(dev, devnode));
-       joysticks.push_back(joystick);
+       CJoystickUdev *j = new CJoystickUdev(dev, devnode);
+       if(j->isInitialized()) {
+	 JoystickPtr joystick = JoystickPtr(j);
+	 joysticks.push_back(joystick);
+       } else {
+	 delete j;
+       }
      }
 
      udev_device_unref(dev);
diff --git a/src/api/udev/JoystickUdev.h b/src/api/udev/JoystickUdev.h
index 6ce5740f..949d2f27 100644
--- a/src/api/udev/JoystickUdev.h
+++ b/src/api/udev/JoystickUdev.h
@@ -58,6 +58,7 @@ namespace JOYSTICK
     virtual bool Initialize(void) override;
     virtual void Deinitialize(void) override;
     virtual void ProcessEvents(void) override;
+    bool isInitialized() { return m_bInitialized; }
 
   protected:
     // implementation of CJoystick
